Lib/test/test_pdb.py

```diff
@@ -2618,13 +2618,29 @@ def _run_pdb(self, pdb_args, commands,
 
     def run_pdb_script(self, script, commands,
                        expected_returncode=0,
-                       extra_env=None):
+                       extra_env=None,
+                       pdbrc=None,
+                       remove_home=False):
         """Run 'script' lines with pdb and the pdb 'commands'."""
         filename = 'main.py'
         with open(filename, 'w') as f:
             f.write(textwrap.dedent(script))
+
+        if pdbrc is not None:
+            with open('.pdbrc', 'w') as f:
+                f.write(textwrap.dedent(pdbrc))
+            self.addCleanup(os_helper.unlink, '.pdbrc')
         self.addCleanup(os_helper.unlink, filename)
-        return self._run_pdb([filename], commands, expected_returncode, extra_env)
+
+        homesave = None
+        if remove_home:
+            homesave = os.environ.pop('HOME', None)
+        try:
+            stdout, stderr = self._run_pdb([filename], commands, expected_returncode, extra_env)
+        finally:
+            if homesave is not None:
```

Comment:
```suggestion
            if remove_home:
```

Reply:
If it was ``None`` before then it should be set back to ``None`` here, no?

Reply:
I don't believe `os.environ` can contain a `None`. It's a mapping object and the key/value are both string https://docs.python.org/3/library/os.html#os.environ .

Reply:
> Do you want me to just revert it?

Revert what?

Reply:
https://github.com/python/cpython/pull/110496/commits/0b1c0e82a023cf7de46b20244757ac074091e1d4 this commit breaks the test because on Windows there's no `HOME` environment. We need to either revert this, or add a `homesave is not None` check after `if remove_home`.

Reply:
Up to you.